File: SpatialGlobalAveragePooling.lua
local SpatialGlobalAveragePooling, Parent = torch.class('nn.SpatialGlobalAveragePooling', 'nn.Module')

function SpatialGlobalAveragePooling:__init()
  Parent.__init(self)
end

function SpatialGlobalAveragePooling:updateOutput(input)
  input.THNN.SpatialGlobalAveragePooling_updateOutput(
      input:cdata(),
      self.output:cdata())    
  return self.output
end

function SpatialGlobalAveragePooling:updateGradInput(input, gradOutput)
  if self.gradInput then
    input.THNN.SpatialGlobalAveragePooling_updateGradInput(
       input:cdata(),
       gradOutput:cdata(),
       self.gradInput:cdata()
       )       
  return self.gradInput
  end
end

File: SpatialGlobalAveragePooling.c
#ifndef TH_GENERIC_FILE
#define TH_GENERIC_FILE "generic/SpatialGlobalAveragePooling.c"
#else

void THNN_(SpatialGlobalAveragePooling_updateOutput)(
          THNNState *state,
          THTensor *input,
          THTensor *output)
{
	printf("SpatialGlobalAveragePooling in CPU is not implemented!");
	exit(-1);
}

void THNN_(SpatialGlobalAveragePooling_updateGradInput)(
          THNNState *state,
          THTensor *input,
          THTensor *gradOutput,
          THTensor *gradInput)
{
	printf("SpatialGlobalAveragePooling in CPU is not implemented!");
	exit(-1);
}

#endif
